hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -3146,3 +3172,54 @@ func globalConfigString(hcluster *hyperv1.HostedCluster) (string, error) {
 func payloadConfigHash(config, targetVersion, pullSecretName, globalConfig string) string {
 	return supportutil.HashSimple(config + targetVersion + pullSecretName + globalConfig)
 }
+
+// mirrorConfigs mirrors higher-level machineConfiguration objects, i.e., kubeletConfig and containerRuntimeConfig, into
+// the HCP namespace, since those are needed as an input for certain operators (such as NTO)
+func (r *NodePoolReconciler) mirrorConfigs(ctx context.Context, logr logr.Logger, mirroredConfigs []corev1.ConfigMap, controlPlaneNamespace string, nodePool *hyperv1.NodePool) error {
+	// get kubeletConfig and containerRuntimeConfig, which already mirrored from the HCP namespace
+	existingConfigsList := &corev1.ConfigMapList{}
+	if err := r.List(ctx, existingConfigsList, &client.ListOptions{
+		Namespace:     controlPlaneNamespace,
+		LabelSelector: labels.SelectorFromValidatedSet(labels.Set{mirroredConfigLabel: ""}),
+	}); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	// need to delete the redundant configs that are no longer part of the nodepool spec
```

Comment:
I would recommend something like the following to make it easy to debug this controller from logs.

```go
want := set.Set[string]{}
for _, configMap := range mirroredConfigs {
    want.Insert(configMap.Name)
}

have := set.Set[string]{}
for _, configMap := range existingConfigsList.Items {
    have.Insert(configMap.Name)
}

toCreate, toDelete := want.Difference(have), have.Difference(want)
if len(toCreate) > 0 {
    logr.WithField("toCreate", toCreate.UnsortedList())
}
if len(toDelete) > 0 {
    logr.WithField("toDelete", toDelete.UnsortedList())
}
if len(toCreate) > 0 || len(toDelete) > 0 {
    logr.Info("updating mirrored configs")
}
```

Reply:
Done.